src/stan_math_backend/Expression_gen.ml

```diff
@@ -138,9 +138,14 @@ let fn_renames =
    hash map of metadata available for each expression that we could put something like this in.
 *)
 let map_rect_counter = ref 0
+
 let functor_suffix = "_functor__"
-let functor_reduce_sum_suffix = "_functor__reduce_sum__"
+let reduce_sum_functor_suffix = "_reduce_sum_functor__"
```

Comment:
This causes a naming conflict the user defines both `foo()` and `foo_reduce_sum()` functions (not very likely, i know).
Could the suffix be simply `"_rsfunctor__"`?

Reply:
Good point. Thanks! Pushing the change.